### ISSUE TITLE
gpt-oss streaming (Ollama /api/chat, Anthropic /v1/messages) silently drops the entire analysis/thinking channel — non-streaming preserves it

### ISSUE THREAD

**Issue by dpalmqvist** [motsognirr/olmlx#713]
## Summary

For gpt-oss (harmony channel-format) models, the model's reasoning/"analysis" channel content is silently discarded on every streaming surface (`/api/chat` stream, `/v1/messages` stream) — never emitted as `message.thinking`, never emitted as a `thinking` content block, and not folded into `content` either. The exact same request in non-streaming mode correctly returns the reasoning text in `message.thinking`. This is a content-loss bug, not just a formatting inconsistency: entire reasoning traces vanish with no trace in the stream.

## Root cause

`_GptOssChannelFilter.should_yield()` (`olmlx/engine/logits_processors.py:242-293`) only returns `True` for tokens in the `final` channel. Tokens in the `analysis` channel are appended to `self._analysis_texts` (line 283) but are **only** surfaced via `get_fallback_texts()` — and that fallback is only drained "if no final channel was produced" (`olmlx/engine/inference.py:3432-3435`). Whenever the model *does* produce both an analysis and a final channel (the normal, common case), the analysis tokens are buffered and then simply dropped — `should_yield` never returns `True` for them, so `yield {"text": token.text, "done": False}` (`inference.py:3415-3418`) is never called for that content, and no downstream router ever sees it.

The non-streaming path is unaffected because it uses `raw_text` (the full unfiltered generation, `inference.py:4201-4222`) and re-parses it with `_parse_gpt_oss_channels`, which correctly extracts the analysis channel into `thinking`.

## Reproduction

Server: `uv run olmlx serve` (http://localhost:11434). Model: `mlx-community/gpt-oss-20b-MXFP4-Q4` (already cached).

**Non-streaming — thinking correctly populated:**
```
curl http://localhost:11434/api/chat -d '{
  "model": "mlx-community/gpt-oss-20b-MXFP4-Q4",
  "messages": [{"role":"user","content":"What is 7 times 8? Think step by step."}],
  "stream": false,
  "options": {"num_predict": 150}
}'
```
Response includes:
```json
"message": {
  "role": "assistant",
  "content": "Sure! Let’s walk through the multiplication step‑by‑step:\n\n1. **Write the numbers**",
  "thinking": "The user asks: \"What is 7 times 8? Think step by step.\" ... Thus final answer: 56. ..."
}
```

**Streaming — thinking is silently empty, no trace of the reasoning anywhere:**
```
curl -N http://localhost:11434/api/chat -d '{
  "model": "mlx-community/gpt-oss-20b-MXFP4-Q4",
  "messages": [{"role":"user","content":"What is 7 times 8? Think step by step."}],
  "stream": true,
  "options": {"num_predict": 150}
}'
```
Every chunk has `"message": {"role": "assistant", "content": "..."}` with no `thinking` key at all; concatenated across the whole stream, accumulated thinking length is **0 characters**. `content` starts directly at `"Sure! Let's walk through..."` — i.e. the reasoning text was fully consumed and discarded upstream of the router, not merely mis-split.

Reproduced identically on a second prompt (`"What is 12 times 13? Think step by step."`), run twice for determinism — both streaming runs: `THINKING LEN: 0`.

**Anthropic streaming — same root cause, same symptom (no `thinking` content block at all):**
```
curl -N http://localhost:11434/v1/messages -d '{
  "model": "mlx-community/gpt-oss-20b-MXFP4-Q4",
  "max_tokens": 150,
  "stream": true,
  "messages": [{"role":"user","content":"What is 9 times 9? Think step by step."}]
}'
```
Only a single `text` content block is streamed (`Sure! Let's walk through the calculation step...`); no `thinking` content block precedes it, even though the Anthropic protocol supports one and the non-streaming Anthropic response for the same kind of request would include it.

## Observed vs Expected

- Observed: streaming loses 100% of the model's reasoning/analysis-channel output on every surface, with no `thinking` field/block and no fallback into `content`.
- Expected: streaming should surface the analysis channel the same way non-streaming does — either as incremental `thinking` chunks (Ollama `message.thinking` deltas, Anthropic `thinking` content blocks) or, at minimum, not silently vanish. Right now a client relying on streaming (the common integration path) never sees any reasoning trace for gpt-oss, while a client using non-streaming gets the full trace — a stream/non-stream data-loss inconsistency, distinct from the already-fixed tag-splitting issues (#306/#471/#659/#686 etc.) since gpt-oss doesn't go through the `<think>`-tag splitter at all — it's filtered out at the token-yield level in `_GptOssChannelFilter`/`_full_completion` before the streaming router code ever runs.

## Relevant log excerpt

```
2026-08-03 17:57:09 [INFO] olmlx.engine.model_manager: [b87254e2] Model mlx-community/gpt-oss-20b-MXFP4-Q4:latest caps: tools=True, thinking=False, thinking_tags=True
2026-08-03 17:57:12 [INFO] olmlx.engine.inference: [b87254e2] Generation complete: 1 prompt tokens (9.8 tok/s), 91 tokens generated (41.9 tok/s), 3.38s total
INFO:     127.0.0.1:60836 - "POST /api/chat HTTP/1.1" 200 OK
```
No errors/exceptions — the requests return clean 200s in both modes, so this is a silent data-loss bug, not a crash.

## Notes

Not the already-documented sampling-determinism behavior (#708) — reproduced with distinct prompts and consistently zero thinking length in streaming across multiple independent runs regardless of prompt.

**Comment by dpalmqvist** [comment]
Confirmed the same root cause also affects the **Responses API** (`/v1/responses`), which the issue title/body doesn't mention — the bug is at the shared `_GptOssChannelFilter.should_yield()` token-yield level (`olmlx/engine/logits_processors.py`), so it's not Ollama/Anthropic-specific.

Non-streaming `/v1/responses` correctly returns a `reasoning` output item with the full `analysis` channel content:
```
curl http://localhost:11434/v1/responses -d '{
  "model": "mlx-community/gpt-oss-20b-MXFP4-Q4",
  "input": "What is 15 * 23? Think step by step."
}'
```
→ `output` includes `{"type": "reasoning", "content": [{"type": "reasoning_text", "text": "We need to respond: ... 345 ..."}]}` followed by the `message` item.

Streaming drops it entirely — no `response.reasoning...` event of any kind is emitted:
```
curl -N http://localhost:11434/v1/responses -d '{
  "model": "mlx-community/gpt-oss-20b-MXFP4-Q4",
  "input": "What is 2+2?",
  "max_output_tokens": 100,
  "stream": true
}'
```
Event sequence is `response.created` → `response.in_progress` → `response.output_item.added` (message) → `response.content_part.added` → `response.output_text.delta` (`"4"`) → `...done` → `response.completed`. `usage.output_tokens: 53` for a 1-token (`"4"`) visible answer confirms ~52 tokens of reasoning were generated and silently discarded, consistent with the `/api/chat` and `/v1/messages` findings above.

Worth updating the issue title/fix scope to cover all three streaming surfaces since the fix is presumably at the shared filter, not per-router.
